src/components/Tables/Table.tsx

```diff
@@ -342,12 +420,33 @@ const Table = <Data,>({
                 <NoSsr>
                     <DataTable
                         columns={columns}
-                        data={toDisplay}
+                        data={rows}
                         keyField="rowId"
                         fixedHeader
-                        pagination={pagination ?? true}
+                        pagination={paginationConfig.pagination}
                         persistTableHead
                         onRowClicked={onRowClick}
+                        paginationServer={paginationConfig.pagination}
+                        paginationTotalRows={
+                            paginationConfig.pagination ? paginationConfig.totalRows : undefined
+                        }
+                        paginationDefaultPage={paginationConfig.pagination ? 1 : undefined}
+                        onChangePage={
+                            paginationConfig.pagination ? paginationConfig.onPageChange : () => {}
+                        }
+                        onChangeRowsPerPage={
+                            paginationConfig.pagination
+                                ? paginationConfig.onPerPageChange
+                                : () => {}
+                        }
+                        sortServer={sortConfig.sortPossible}
+                        onSort={handleSort}
+                        progressComponent={
+                            <Centerer>
+                                <CircularProgress aria-label="table-progress-bar" />
+                            </Centerer>
+                        }
+                        progressPending={isLoading}
```

Comment:
the progress component shows within the table in place of the data whenever the data is loading. This is different from rendering a progress component instead of the table in the parent page whenever the data is loading, because this way the table states are maintained.